File: Beecrowd_Parte3/O_Jardin_3148.c
#include <stdio.h>
#include <stdlib.h>

// Função de comparação para ordenação decrescente
int compare(const void *a, const void *b) {
    return (*(int *)b - *(int *)a);
}

int main() {
    int t;
    scanf("%d", &t); // número de casos de teste
    
    while (t--) {
        int n;
        scanf("%d", &n); // número de coretos
        
        int X[n];
        for (int i = 0; i < n; i++) {
            scanf("%d", &X[i]); // leitura das quantidades de vírus em cada coreto
        }
        
        // Ordena o vetor X em ordem decrescente
        qsort(X, n, sizeof(int), compare);
        
        // Calcula o número máximo de vírus que podem ser eliminados
        long long totalVirusEliminated = 0;
        for (int i = 0; i < n; i++) {
            totalVirusEliminated += (long long)X[i] * (n - i);
        }
        
        // Imprime o resultado para o caso de teste atual
        printf("%lld\n", totalVirusEliminated);
    }
    
    return 0;
}

File: Beecrowd_Parte3/O_Jardin_3148.java
import java.util.*;

public class O_Jardin_3148 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        int t = sc.nextInt(); // número de casos de teste
        
        while (t-- > 0) {
            int n = sc.nextInt(); // número de coretos
            
            // Lê o vetor X de quantidades de vírus em cada coreto
            int[] X = new int[n];
            for (int i = 0; i < n; i++) {
                X[i] = sc.nextInt();
            }
            
            // Ordena o vetor X em ordem decrescente
            Arrays.sort(X);
            
            // Calcula o número máximo de vírus que podem ser eliminados
            long totalVirusEliminated = 0;
            for (int i = 0; i < n; i++) {
                totalVirusEliminated += (long) X[i] * (n - i);
            }
            
            // Imprime o resultado para o caso de teste atual
            System.out.println(totalVirusEliminated - 3);
        }
        
        sc.close();
    }
}
